### PR TITLE
Add command validation to utils and ff command

### PR DESCRIPTION
**Pull request by saschagrunert**
We now introduce a new utility function to verify that system commands
are available. Unit tests have been added as well.

### CODE REVIEW

**Comment by justaugustus** [comment]
@saschagrunert -- I think if you'll need to run `./hack/update-all.sh` and then `./hack/verify-all.sh` to make sure the bazel file gets rebuilt correctly.

File: cmd/krel/cmd/ff.go [hunk 1]
```diff
@@ -77,11 +77,14 @@ func init() {
 
 func runFf(opts *Options) error {
 	// TODO: Add usage/help
-	// TODO: Check prerequisites (git, jq, go, make)
 	// TODO: Set positional args
 	// TODO: Fail on empty branch
 	// TODO: Fail on GITHUB_TOKEN not set
 
+	if !util.CommandsAvailable([]string{"git", "go", "make", "jq"}) {
```
Comment (hoegaarden) [inline cmd/krel/cmd/ff.go]: QQ here:
I thought `krel` uses go-git? Why do we have a dependency on git here too? Can we maybe only depend on either one?

Comment (justaugustus) [reply cmd/krel/cmd/ff.go]: There are certain git commands that we use that are not supported in `go-git`.
Here's the compatibility list: https://github.com/src-d/go-git/blob/master/COMPATIBILITY.md

**Pull request merged**